### ISSUE TITLE
Reality+H2 `(httpSettings)`

### ISSUE THREAD

**Issue by hawshemi** [chika0801/Xray-examples#1]
Hi
I have a question regarding the reality+h2 config settings.

In the [Docs](https://xtls.github.io/en/config/transports/h2.html#httpobject) it is mentioned about `httpSettings` but in your `server_config.json` there is no section about it. 

And what is this object exactly?
```
{
  "host": ["xray.com"],
  "path": "/random/path",
  "read_idle_timeout": 10,
  "health_check_timeout": 15,
  "method": "PUT",
  "headers": {
    "Header": ["value"]
  }
}
```
Let's say my `server_config.json` is this:
```
{
    "log": {
        "loglevel": "warning"
    },
    "routing": {
        "domainStrategy": "IPIfNonMatch",
        "rules": [
            {
                "type": "field",
                "port": "443",
                "network": "udp",
                "outboundTag": "block"
            }
        ]
    },
    "inbounds": [
        {
            "listen": "0.0.0.0",
            "port": 443,
            "protocol": "vless",
            "settings": {
                "clients": [
                    {
                        "id": "uuId"
                    }
                ],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "h2",
                "security": "reality",
                "realitySettings": {
                    "dest": "python.org:443",
                    "serverNames": [
                        "python.org", "www.python.org"
                    ],
                    "privateKey": "privKey", 
                    "shortIds": [ 
                        "sId"
                    ]
                },
                "httpSettings": {
                  "host": ["python.org"],
                  "path": "/ftp/python",
                  "read_idle_timeout": 10,
                  "health_check_timeout": 15,
                  "method": "PUT"
                }
            },
            "sniffing": {
                "enabled": true,
                "destOverride": [
                    "http",
                    "tls",
                    "quic",
                    "fakedns"
                ]
            }
        }
    ],
    "outbounds": [
        {
            "protocol": "freedom",
            "tag": "direct"
        },
        {
            "protocol": "blackhole",
            "tag": "block"
        }
    ],
    "policy": {
        "levels": {
            "0": {
                "handshake": 2,
                "connIdle": 120
            }
        }
    }
}
```
Is this section:
```
                "httpSettings": {
                  "host": ["python.org"],
                  "path": "/ftp/python",
                  "read_idle_timeout": 10,
                  "health_check_timeout": 15,
                  "method": "PUT"
                }
```
Doing anything?

And my next question is do you know what is the `headers` in the object?
```
"headers": {
    "Header": ["value"]
  }
```

**Comment by chika0801** [comment]
> And my next question is do you know what is the headers in the object?

This is a parameter that I never use in my own normal use. I'm also not sure in what scenarios it is used.

```
                "httpSettings": {
                  "host": ["python.org"],
                  "path": "/ftp/python",
                  "read_idle_timeout": 10,
                  "health_check_timeout": 15,
                  "method": "PUT"
                }
```

`"host"`

This one parameter, if you don't write it, has a default value of www.example.com. This default value is a convention from within the v2ray code. Nowadays, v2fly and xray also inherits the value www.example.com if you don't write this parameter. sing-box core, if you don't write this parameter too. sing-box also follows this tradition and uses www.example.com.

However, for some clients like Shadowrocket, you need to manually fill in www.example.com in the Host item in order for the client to connect to the server.

To put it another way, the value of the Host item needs to be the same for both the server and client configurations. host is similar to the server name in gRPC.

```
                  "read_idle_timeout": 10,
                  "health_check_timeout": 15,
```

These 2 are http2 health check parameters, there may be some time to the phenomenon of broken flow, can be solved.

What the other parameters do I don't know.

**Comment by hawshemi** [comment]
So these settings are regarding the `network": "h2",` ?
Do you know what the difference is between a specified host and path and the default one?
Do you know what it does? because I didn't find anything on the v2ray-core docs about it.

**Comment by chika0801** [comment]
> So these settings are regarding the network": "h2", ?
```
                "httpSettings": {
                  "host": ["python.org"],
                  "path": "/ftp/python",
                  "read_idle_timeout": 10,
                  "health_check_timeout": 15,
                  "method": "PUT"
                }
```
The above parameters are the use of http2 below the subsidiary parameters, is optional, you do not fill in the default values used.


> Do you know what the difference is between a specified host and path and the default one?

`"path": "/ftp/python",` This parameter I have never used it. So I don't know where it should be used.

> Do you know what it does? because I didn't find anything on the v2ray-core docs about it.

` "host":`You just need to keep an eye on this parameter, if you specify its value, it must be the same on the client side as on the server side for the connection to succeed.

The client can additionally use the health check for those 2 parameters.

Other points to note I think are missing.

**Comment by hawshemi** [comment]
Thanks for the response.
